### ISSUE TITLE
AWS::IAM::ManagedPolicy Condition syntax error not detected

### ISSUE THREAD

**Issue by gbudge** [aws-cloudformation/cfn-lint#3751]
### CloudFormation Lint Version

cfn-lint 1.16.1

### What operating system are you using?

Linux

### Describe the bug

cfn-lint is not detecting missing Condition operator.

```shell
$ cfn-lint --format json --include-checks I -t ./demo-template.yaml
[]

$ 
```

CloudFormation throws the following syntax error:

```
Resource handler returned message: "Syntax errors in policy. (Service: Iam, Status Code: 400, Request ID: <redacted>)" (RequestToken: <redacted>, HandlerErrorCode: InvalidRequest)
```

### Expected behavior

The missing operator should be detected.

### Reproduction template

```yaml
---
AWSTemplateFormatVersion: '2010-09-09'
Description: Example IAM policy syntax error not detected.

Resources:
  IAMPolicy:
    Type: AWS::IAM::ManagedPolicy
    Properties:
      Roles:
        - DemoRole
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action:
              - servicecatalog:*
            Resource: '*'
            Condition:
                servicecatalog:accountLevel: self
```
